test/helpers/ProposalTest.sol

```diff
@@ -0,0 +1,336 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.20;
+
+import {ICompoundTimelock} from
+  "@openzeppelin/contracts/governance/extensions/GovernorTimelockCompound.sol";
+import {IERC20} from "@openzeppelin/contracts/interfaces/IERC20.sol";
+import {IGovernor} from "@openzeppelin/contracts/governance/IGovernor.sol";
+
+import {TestableProposeScript} from "test/helpers/TestableProposeScript.sol";
+import {IGovernorAlpha} from "src/interfaces/IGovernorAlpha.sol";
+import {RadworksGovernorTest} from "test/helpers/RadworksGovernorTest.sol";
+
+abstract contract ProposalTest is RadworksGovernorTest {
+  //----------------- State and Setup ----------- //
+
+  IGovernorAlpha governorAlpha = IGovernorAlpha(GOVERNOR_ALPHA);
+  ICompoundTimelock timelock = ICompoundTimelock(payable(TIMELOCK));
+  uint256 initialProposalCount;
+  uint256 upgradeProposalId;
+
+  // As defined in the GovernorAlpha ProposalState Enum
+  uint8 constant PENDING = 0;
+  uint8 constant ACTIVE = 1;
+  uint8 constant DEFEATED = 3;
+  uint8 constant SUCCEEDED = 4;
+  uint8 constant QUEUED = 5;
+  uint8 constant EXECUTED = 7;
+
+  // From GovernorCountingSimple
+  uint8 constant AGAINST = 0;
+  uint8 constant FOR = 1;
+  uint8 constant ABSTAIN = 2;
+
+  function setUp() public virtual override {
+    RadworksGovernorTest.setUp();
+
+    if (_useDeployedGovernorBravo()) {
+      // Use the actual live proposal data when it gets put on chain
```

Comment:
can you help me understand here, why will the next proposal Id be 17?

the "when it gets put on chain" part of the comment is a little confusing. assumedly if we're inside this logic branch, it's already on chain.

Reply:
Done: d21c155b4535c2cda958551692d4caa1ef01bd1e